packages/shared/utils.ts

```diff
@@ -33,3 +33,8 @@ export function merge(target: any, source: any) {
 export function isIndex(value: unknown): value is number {
   return Number(value) >= 0;
 }
+
+export function toNumber(value: string): number | string {
+  const n = parseFloat(value);
+  return isNaN(n) ? value : n;
```

Comment:
Minor issue, I prefer to keep a new line before `return` statements
```suggestion

  return isNaN(n) ? value : n;
```

Reply:
Done. In this case, I would suggest adding an ESLint rule for this.

See: https://eslint.org/docs/rules/newline-before-return